l10n_es_aeat/hooks.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Acsone SA - Xavier Bouquiaux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import tools
+from odoo.tools.sql import column_exists
+
+_logger = logging.getLogger(__name__)
+
+
+def create_column_thirdparty_invoice(cr):
+    if not column_exists(
+        cr, "account_journal", "thirdparty_invoice"
+    ) and not column_exists(cr, "account_move", "thirdparty_invoice"):
+        _logger.info("Initializing column thirdparty_invoice on table account_move")
+        tools.create_column(
+            cr=cr,
+            tablename="account_move",
+            columnname="thirdparty_invoice",
+            columntype="boolean",
+            comment="Third-party invoice",
+        )
+        tools.create_column(
+            cr=cr,
+            tablename="account_journal",
+            columnname="thirdparty_invoice",
+            columntype="boolean",
+            comment="Third-party invoice",
+        )
+        cr.execute(
+            """
+            UPDATE account_move
+            SET thirdparty_invoice = data_rec.thirdparty_invoice
+
+            from  ( SELECT am.id as am_id,
+                            aj.thirdparty_invoice as thirdparty_invoice
+                FROM account_move am
+                JOIN account_journal aj on aj.id = am.journal_id
+                WHERE aj.thirdparty_invoice IS NOT NULL ) data_rec
+
+
+            where data_rec.am_id = account_move.id
```

Comment:
You don't need this, as if entering this part, the column didn't exist in account_journal, so it will always be False. You can do:

```suggestion
UPDATE account_move SET thirdparty_invoice = False
```

Doing some exercising, if the column exists in account_journal, the simplified SQL would be:

```python
            UPDATE account_move
            SET thirdparty_invoice = aj.thirdparty_invoice
            JOIN account_journal
            WHERE aj.id = am.journal_id
```

Reply:
Done
Thanks for the feedback @pedrobaeza